src/modules/flow/http-client/http-client.c

```diff
@@ -0,0 +1,702 @@
+        sol_http_param_free(&params);
+        return -ENOMEM;
+    }
+
+    connection = sol_http_client_request(SOL_HTTP_METHOD_GET, mdata->url,
+        &params, boolean_request_finished, mdata);
+
+    sol_http_param_free(&params);
+
+    SOL_NULL_CHECK(connection, -ENOTCONN);
+
+    r = sol_ptr_vector_append(&mdata->pending_conns, connection);
+    if (r < 0) {
+        SOL_WRN("Failed to keep pending connection.");
+        sol_http_client_connection_cancel(connection);
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+static int
+boolean_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    bool b;
+    struct sol_http_param params;
+    struct http_data *mdata = data;
+    struct sol_http_client_connection *connection;
+
+    r = sol_flow_packet_get_boolean(packet, &b);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_http_param_init(&params);
+    if (!(sol_http_param_add(&params,
+        SOL_HTTP_REQUEST_PARAM_HEADER("Accept", "application/json"))) ||
+        !(sol_http_param_add(&params,
+        SOL_HTTP_REQUEST_PARAM_POST_FIELD("value", b ? "true" : "false")))) {
+        SOL_WRN("Failed to set query params");
+        sol_http_param_free(&params);
+        return -ENOMEM;
+    }
+    connection = sol_http_client_request(SOL_HTTP_METHOD_POST, mdata->url,
+        &params, boolean_request_finished, mdata);
+    sol_http_param_free(&params);
+
+    SOL_NULL_CHECK(connection, -ENOTCONN);
+
+    r = sol_ptr_vector_append(&mdata->pending_conns, connection);
+    if (r < 0) {
+        SOL_WRN("Failed to keep pending connection.");
+        sol_http_client_connection_cancel(connection);
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+static int
+boolean_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_client_boolean_options *opts =
+        (struct sol_flow_node_type_http_client_boolean_options *)options;
+    mdata->node = node;
+
+    return common_open(mdata, opts->url);
+}
+
+
+/*
+ * --------------------------------- string node -----------------------------
+ */
+static void
+string_request_finished(void *data,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    struct http_data *mdata = data;
+    char *result = NULL;
+
+    if (check_response(mdata, connection, response) < 0)
+        return;
+
+    if (streq(response->content_type, "application/json")) {
+        struct sol_json_scanner scanner;
+        struct sol_json_token token, key, value;
+        enum sol_json_loop_reason reason;
+
+        sol_json_scanner_init(&scanner, response->content.data, response->content.used);
+        SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+            result = strndup(value.start + 1, value.end - value.start - 2);
```

Comment:
indeed, if more keys were given, then it would leak. As for boolean, we should check for a specific key.

Also we shouldn't duplicate the memory and for JSON we do have a call to remove quotes and I've asked @bdilly to add one to also handle unescape of the string if needed (ie if it contains a `\'`).